### PR TITLE
Add BitTiming/BitTimingFd support to KvaserBus

### PR DESCRIPTION
**Pull request by zariiii9003**
Can someone with a Kvaser device test KvaserBus instantiation with BitTiming and BitTimingFd?

### CODE REVIEW

**Comment by zariiii9003** [comment]
@lumagi I'd like to include this one in 4.3. We'll start with a release candidate anyway. Could you take a look?

File: can/interfaces/kvaser/canlib.py [hunk 1]
```diff
@@ -468,22 +493,43 @@ def __init__(self, channel, can_filters=None, **kwargs):
...
+                    bitrate = BITRATE_FD[bitrate]
+                if data_bitrate in BITRATE_FD:
+                    # Use predefined bitrate for data
+                    data_bitrate = BITRATE_FD[data_bitrate]
+                elif not data_bitrate:
+                    # Use same bitrate for arbitration and data phase
+                    data_bitrate = bitrate
+                canSetBusParamsFd(self._read_handle, data_bitrate, tseg1, tseg2, sjw)
+            else:
+                if "tseg1" not in kwargs and bitrate in BITRATE_OBJS:
```
Comment (lumagi) [inline can/interfaces/kvaser/canlib.py:530]: This is unrelated to your PR, but do they always use the same `tseg1` / `tseg2` for both nominal and data phases?

**Review by lumagi — APPROVED** [review]
Looks very fine to me. I think the legacy code paths should be executed just fine in case no timing parameter is passed.

File: can/interfaces/kvaser/canlib.py [hunk 2]
```diff
@@ -468,22 +493,43 @@ def __init__(self, channel, can_filters=None, **kwargs):
...
-            canSetBusParamsFd(self._read_handle, data_bitrate, tseg1, tseg2, sjw)
+        if isinstance(timing, BitTimingFd):
+            timing = check_or_adjust_timing_clock(timing, [80_000_000])
+            canSetBusParams(
+                self._read_handle,
+                timing.nom_bitrate,
+                timing.nom_tseg1,
+                timing.nom_tseg2,
+                timing.nom_sjw,
+                1,
```
[the remark this answers is not in the document]
Comment (zariiii9003) [reply can/interfaces/kvaser/canlib.py:504]: Does that setting even do anything for CAN FD? I cannot find it in ISO 11898 at all. I'm not sure what the official name for that feature is...
But yeah, i'll change the default value to 1.

File: can/interfaces/kvaser/canlib.py (hunk 1 above)
Comment (zariiii9003) [reply can/interfaces/kvaser/canlib.py:530]: My guess is, if you use the predefined constants (e.g. `canFD_BITRATE_500K_80P = -1000`), then these parameters are irrelevant and the sample point will always be at 80%

File: can/interfaces/kvaser/canlib.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zariiii9003) [reply can/interfaces/kvaser/canlib.py:504]: The CI failed so i changed it back to zero until someone complains or we remove the parameters entirely.

[the remark this answers is not in the document]
Comment (lumagi) [reply can/interfaces/kvaser/canlib.py:504]: Yeah, it looks like 0 is hardcoded in the test case mocks too. It's hard to test this without actual hardware :-/

**Pull request merged**